### PR TITLE
UploadComplete: Use RequestContext getUser to fetch user

### PR DESCRIPTION
**Pull request by paladox**
Rather than relying on fetching it from the file, instead use the current request.

Bug: T12339

### CODE REVIEW

**Review by BlankEclair — COMMENTED** [review]
Can we lookup the global central auth ID before creating the job?

**Comment by paladox** [comment]
> Can we lookup the global central auth ID before creating the job?

What do you mean @BlankEclair ?

**Comment by BlankEclair** [comment]
Instead of doing it in GlobalNewFilesInsertJob, what about doing it in onUploadComplete? I don't know the intricacies of the job queue, but I believe that it has to save the User object (or a serialization of that) somewhere? If so, why not save only the int of the CentralAuth ID, which should be smaller

(This is kind of a hyperoptimization, but I don't see why not)

**Comment by paladox** [comment]
> Instead of doing it in GlobalNewFilesInsertJob, what about doing it in onUploadComplete? I don't know the intricacies of the job queue, but I believe that it has to save the User object (or a serialization of that) somewhere? If so, why not save only the int of the CentralAuth ID, which should be smaller
> 
> (This is kind of a hyperoptimization, but I don't see why not)

oh you mean doing `$centralIdLookup->centralIdFromLocalUser` in onUploadComplete instead of the job

**Comment by paladox** [comment]
@BlankEclair done

File: includes/GlobalNewFilesHooks.php [hunk 1]
```diff
@@ -43,8 +43,12 @@ public static function onCreateWikiStatePublic( $dbname ) {
 	}
 
 	public static function onUploadComplete( $uploadBase ) {
+		$user = RequestContext::getMain()->getUser();
+		$services = MediaWikiServices::getInstance();
+		$centralIdLookup = $services->getCentralIdLookup();
+		$userId = $centralIdLookup->centralIdFromLocalUser( $user );
 		MediaWikiServices::getInstance()->getJobQueueGroup()->push(
```
Comment (BlankEclair) [inline includes/GlobalNewFilesHooks.php:50]: Redundantly retrieves MediaWikiServices twice

Comment (paladox) [reply includes/GlobalNewFilesHooks.php]: Thanks, done.

**Comment by paladox** [comment]
@BlankEclair done

File: includes/GlobalNewFilesHooks.php [hunk 2]
```diff
@@ -43,8 +43,12 @@ public static function onCreateWikiStatePublic( $dbname ) {
 	}
 
 	public static function onUploadComplete( $uploadBase ) {
-		MediaWikiServices::getInstance()->getJobQueueGroup()->push(
-			new GlobalNewFilesInsertJob( $uploadBase->getTitle(), [] )
+		$user = RequestContext::getMain()->getUser();
+		$services = MediaWikiServices::getInstance();
+		$centralIdLookup = $services->getCentralIdLookup();
+		$userId = $centralIdLookup->centralIdFromLocalUser( $user );
```
Comment (BlankEclair) [inline includes/GlobalNewFilesHooks.php:49]: Technically could all go in one expression, but it's minor enough that I don't care

**Review by BlankEclair — APPROVED** [review]

**Comment by paladox** [comment]
Thanks, done.

**Pull request merged**